azurelinuxagent/ga/remoteaccess.py

```diff
@@ -0,0 +1,163 @@
+        self.protocol_util = get_protocol_util()
+        self.protocol = None
+        self.os_util = None
+        self.cryptUtil = CryptUtil(conf.get_openssl_cmd())
+        self.remote_access = None
+
+    def run(self):
+        try:
+            if self.os_util is None:
+                self.os_util = get_osutil()
+            if self.os_util.jit_enabled:
+                logger.verbose("Handle remote access updates")
+                self.protocol = self.protocol_util.get_protocol()
+                self.protocol.client.update_goal_state(True)
+                self.protocol.client.update_remote_access_conf(self.protocol.client.goal_state)
+                if self.remote_access is None or self.remote_access.incarnation != self.protocol.client.remote_access.incarnation:
+                    self.remote_access = self.protocol.client.remote_access
+                    self.handle_remote_access()
+            else:
+                logger.info("Non JIT enabled client.")
+        except Exception as e:
+            msg = u"Exception processing remote access handler: {0}".format(
+                ustr(e))
+            logger.warn(msg)
+            logger.info(str(e))
+            add_event(AGENT_NAME,
+                      version=CURRENT_VERSION,
+                      op=WALAEventOperation.HandleRemoteAccess,
+                      is_success=False,
+                      message=msg)
+            return     
+
+    def handle_remote_access(self):
+        logger.verbose("Entered handle_remote_access")        
+        if self.remote_access is not None:
+            # Get JIT user accounts.
+            all_users = self.os_util.get_users()
+            jit_users = set()
+            for usr in all_users:
+                if usr[4] == REMOTE_ACCESS_ACCOUNT_COMMENT:
+                    jit_users.add(usr[0])     
+            for acc in self.remote_access.user_list.users:
+                raw_expiration = acc.expiration
+                account_expiration = datetime.strptime(raw_expiration, REMOTE_USR_EXPIRATION_FORMAT)
+                now = datetime.utcnow()
+                if acc.name not in jit_users and now < account_expiration:
+                    self.add_user(acc.name, acc.encrypted_password, account_expiration)
+
+    def add_user(self, username, encrypted_password, account_expiration, retry=MAX_TRY_ATTEMPT, throttle=FAILED_ATTEMPT_THROTTLE):
+        created = False
+        attempt_count = 0
+        while attempt_count < retry:
+            try:
+                attempt_count += 1
+                expiration_date = account_expiration + timedelta(days=1) - datetime.utcnow()
+                logger.verbose("add_user: Adding user {0} with expiration in {1} day(s)".format(username, expiration_date.days))
+                self.os_util.useradd(username, expiration_date.days, REMOTE_ACCESS_ACCOUNT_COMMENT)
+                created = True
+                cache_file = os.path.join(conf.get_lib_dir(), "remote_access_pwd.dat")
```

Comment:
>cache_file = os.path.join(conf.get_lib_dir(), "remote_access_pwd.dat") [](start = 16, length = 70)

This needs to be a temporary file.  It should be created inside the decrypt method if you need to have a file.

Reply:
Ideally I'll remove the temp file completely, but if not for some reason, I will move this inside. 

---
In reply to: [192217352](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r192217352) [](ancestors = 192217352)

Reply:
temp file logic removed

---
In reply to: [192223247](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r192223247) [](ancestors = 192223247,192217352)